_pytest/main.py

```diff
@@ -175,7 +175,7 @@ def pytest_ignore_collect(path, config):
     if excludeopt:
         ignore_paths.extend([py.path.local(x) for x in excludeopt])
 
-    if path in ignore_paths:
+    if py.path.local(path) in ignore_paths:
```

Comment:
p is a path.dirpath(), but not path. 
I removed the local variable `p` to make it clearer